tests/database/test_db.py

```
@@ -0,0 +1,1290 @@
+        mock_config_class.db.synchronous_mode = "NORMAL"
+        mock_config_instance.db.as_dict.return_value = {"name": "/tmp/test.db"}
+        mock_config_instance.settings.timeout = 30
+
+        mock_connection = Mock()
+        mock_cursor = Mock()
+        mock_connection.cursor.return_value = mock_cursor
+        mock_apsw_conn.return_value = mock_connection
+
+        result = create_connection()
+
+        mock_apsw_conn.assert_called_once_with("/tmp/test.db")
```

Ruff (0.12.2)

Probable insecure usage of temporary file or directory: "/tmp/test.db"

(S108)